### PR TITLE
Add procedure to change attribute aspect

### PR DESCRIPTION
**Pull request by kkraune**
FYI @geirst 

document the relatively common procedure to migrate from attribute to index (or the other way) safely, using the new reindex features

### CODE REVIEW

File: en/operations/admin-procedures.html [hunk 1]
```diff
@@ -121,6 +121,217 @@ <h2 id="status-pages">Status pages</h2>
...
+
+<h3 id="attribute-to-index">Attribute to index</h3>
+<p>
+  Changing between <code>index</code> and <code>attribute</code> is a common field change operation
+  to optimize performance.
+  Use the <a href="../reindexing.html">reindexing</a> feature to safely migrate data to/from index structures.
+</p>
+<p>
+  When the attribute aspect of a field is removed, the attribute data is dropped on next process restart.
+  Hence, doing a direct change from attribute to index is risky, as a node stop can incur field data loss.
```
Comment (jonmv) [inline en/operations/admin-procedures.html:140]: 😱

File: en/operations/admin-procedures.html [hunk 2]
```diff
@@ -121,6 +121,217 @@ <h2 id="status-pages">Status pages</h2>
...
+</p>
+<pre>
+$ curl http://localhost:19071/application/v2/tenant/default/application/default/environment/prod/region/default/instance/default/reindex
+
+{
+  "generation": 3,
+  "applicationPackageFileReference": "",
+  "modelVersions": ["7.423.37"]
+}
+
```
Comment (jonmv) [inline en/operations/admin-procedures.html:257]:
```suggestion
$ curl http://localhost:19071/application/v2/tenant/default/application/default/environment/prod/region/default/instance/default/reindex

{
  "generation": 3,
  "applicationPackageFileReference": "",
  "modelVersions": ["7.423.37"]
}

```

This seems to match something completely irrelevant? That path is used with POST/DELETE to (de)activate reindexing, AFAIR, but isn't documented.

File: en/operations/admin-procedures.html (hunk 1 above)
Comment (bratseth) [reply en/operations/admin-procedures.html:140]: Not really data loss though, but a gap in availability?

File: en/operations/admin-procedures.html [hunk 3]
```diff
@@ -121,6 +121,217 @@ <h2 id="status-pages">Status pages</h2>
...
+      },
+      "ready": {
+        "music": {}
+      }
+    }
+  }
+}
+
+$ (cd src/main/application && zip -r - .) |   curl --header Content-Type:application/zip --data-binary @- \
+  localhost:19071/application/v2/tenant/default/prepareandactivate
```
Comment (jonmv) [inline en/operations/admin-procedures.html:274]: Hmm, why not use `vespa-deploy`? I thought that was the most common use? The `/application/v2/.../reindexing` API was intended for manually triggering reindexing, when using that deploy tool. `prepareandactivate` indeed lets the change be automatically detected, and acted upon, but I don't think that's even in the opensource documentation—neither of them?

Comment (jonmv) [reply en/operations/admin-procedures.html:274]: ... not saying it shouldn't be, though.

File: en/operations/admin-procedures.html [hunk 4]
```diff
@@ -121,6 +121,217 @@ <h2 id="status-pages">Status pages</h2>
...
+      }
+    }
+  }
+}
+
+$ docker exec vespa /usr/bin/sh -c '/opt/vespa/bin/vespa-logfmt | grep Reindexer'
+[2021-08-06 09:13:00.524] INFO    : container-clustercontroller Container.ai.vespa.reindexing.Reindexer
+                                    Completed reindexing of datatype music (code: 1412693671) after PT0.547121S
+
+$ docker exec vespa /usr/bin/sh -c '/opt/vespa/bin/vespa-proton-cmd --local triggerFlush' && \
```
Comment (jonmv) [inline en/operations/admin-procedures.html:298]: What/why is this?

**Review by jonmv — COMMENTED** [review]
Much great information here. I do have some comments, though.

File: en/operations/admin-procedures.html (hunk 1 above)
Comment (kkraune) [reply en/operations/admin-procedures.html:140]: will test again to qualify this, could be I did this wrong

**Comment by kkraune** [comment]
talked with Jon Marius, clarified a lot, v2 of this procedure coming shortly

File: en/operations/admin-procedures.html (hunk 3 above)
Comment (kkraune) [reply en/operations/admin-procedures.html:274]: Using `prepareandactivate` makes it more similar to Vespa Cloud, easier to swap between the two. See https://docs.vespa.ai/en/cloudconfig/deploy-rest-api-v2.html#prepareandactivate . This doc needs to be updated as it is more than a prepare/activate (as it actually will do reindex), will do this soon

File: en/operations/admin-procedures.html (hunk 4 above)
Comment (kkraune) [reply en/operations/admin-procedures.html:298]: I added this to be able to see the effects of the indexing changes - I think I will move this to an appendix instead, for debugging purposes, not really needed here

**Comment by kkraune** [comment]
@jonmv updated after we talked yesterday, the procedure now works without the reindex call that was broken anyway.

I will separate this procedure into an own doc and add some more details / have a separate troubleshooting section etc. But would like to merge this in for now.

Finally, will test again to see if the node actually drops data or just temporarily

**Review by jonmv — APPROVED** [review]
Nice!

File: en/operations/admin-procedures.html (hunk 3 above)
Comment (kkraune) [reply en/operations/admin-procedures.html:291]: ok - so maybe we need a new deploy api command then as https://docs.vespa.ai/en/cloudconfig/deploy-rest-api-v2.html requires a session (unless one can use * or something as session ID to use latest / create one auto)..

We can discuss that separately, I'll create a ticket

Comment (jonmv) [reply en/operations/admin-procedures.html:291]: > activate without prepare does an implicit prepare

I don't think that's true, for the HTTP API—looking at the code, the activate endpoint explicitly creates a `Deployment.prepared(...)`, which is then activated.

File: en/operations/admin-procedures.html (hunk 1 above)
Comment (kkraune) [reply en/operations/admin-procedures.html:140]: you are right, @bratseth  - the data survived a proton restart even though not yet reindexed from attribute into index. after subsequent deployment, which triggered the actual reindex, the index was populated. Will rewrite to availability not loss in next PR. Thanks!

**Pull request merged**